nerfstudio/model_components/renderers.py

```diff
@@ -335,47 +335,46 @@ def forward(
         ray_samples: RaySamples,
         ray_indices: Optional[Int[Tensor, "num_samples"]] = None,
         num_rays: Optional[int] = None,
-    ) -> Float[Tensor, "*batch 1"]:
+        quantity: Optional[Float[Tensor, "*batch num_samples 1"]] = None,
+    ) -> Float[Tensor, "*bs 1"]:
         """Composite samples along ray and calculate depths.
 
         Args:
             weights: Weights for each sample.
             ray_samples: Set of ray samples.
             ray_indices: Ray index for each sample, used when samples are packed.
             num_rays: Number of rays, used when samples are packed.
+            quantity: If specified, then take the median of this quantity instead of the depth.
 
         Returns:
             Outputs of depth values.
         """
 
-        if self.method == "median":
-            steps = (ray_samples.frustums.starts + ray_samples.frustums.ends) / 2
+        if quantity is None:
```

Comment:
move median logic to function -> top of file.
Create a new renderer called VisibilityRenderer -> only does median depth.